### PR TITLE
refactor: combine typst-preview and tinymist extension code

### PR DESCRIPTION
**Pull request by Myriad-Dreamin**
Previous PRs:
- #337
- #332 
- #323 

---

Start to merge extension code, and we make some "compatible" branch to override logic.
Since this PR, you don't have to install two extensions and all preview configurations and commands are *moved* to `myriaddreamin.tinymist`.

---

The only two non-trivial changes of `mgt19937.typst-preview`:
- override `getCliPath` by `tinymist.getServer`
  ```ts
  export async function getPreviewCliPath(extensionPath?: string): Promise<string> {
      if (tinymistServerConfig) {
          return getServer(tinymistServerConfig!);
      }
      ...}
  ```
- remove `typst-guy`. since we will have a unique status indicator in end.
    ```ts
  let guy = "$(typst-guy)";
  export async function setIsTinymist(config: Record<string, any>) {
      guy = "$(sync)";
  }
  
    ```

### CODE REVIEW

**Review by Enter-tainer — APPROVED** [review]

File: editors/vscode/package.json [hunk 1]
```diff
@@ -40,6 +40,17 @@
                     "type": "webview",
                     "name": "Symbol View",
                     "when": "resourceLangId == typst"
+                },
+                {
+                    "id": "typst-preview.content-preview",
```
[the remark this answers is not in the document]
Comment (Myriad-Dreamin) [reply editors/vscode/package.json:45]: If we don't rename any of them, people just need to uninstall the old extension and no configuration requires to change.

I think we can give some time window to move configurations, we may deprecate them and prefer to use tinymist configurations first. For example:
- `tinymist.fontPaths` v.s. `typst-preview.fontPaths`.
- `tinymist.preview.invertColors` v.s. `typst-preview.invertColors`.

Luckily, we don't have to do them at once.

File: editors/vscode/src/preview-compat.ts [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+    statusBarItem.tooltip = "Typst Preview Status: Click to show logs";
+    return statusBarItem;
+}
+
+export function previewStatusBarItemProcess(
+    event: "Compiling" | "CompileSuccess" | "CompileError"
+) {
+    // if (isTinymist) {
+    //     return;
+    // }
```
[the remark this answers is not in the document]
Comment (Enter-tainer) [reply editors/vscode/src/preview-compat.ts:98]: make sense

File: contrib/typst-preview/editors/vscode/package.json [hunk 3]
```diff
@@ -252,12 +254,13 @@
   },
   "scripts": {
     "build:frontend": "cd ../../../../tools/typst-preview-frontend && yarn run build && cd ../../contrib/typst-preview/editors/vscode && rimraf ./out/frontend/ && cpr ../../../../tools/typst-preview-frontend/dist/ ./out/frontend/ && rimraf ../../../../crates/tinymist-assets/src/typst-preview.html && cpr ./out/frontend/index.html ../../../../crates/tinymist-assets/src/typst-preview.html",
-    "vscode:prepublish": "yarn run compile",
-    "compile": "tsc -p ./ && yarn run build:frontend",
+    "build-base": "esbuild ../../../../editors/vscode/src/extension.preview.ts --bundle --outfile=out/extension.js --external:vscode --format=cjs --platform=node --target=node16",
```
[the remark this answers is not in the document]
Comment (Myriad-Dreamin) [reply contrib/typst-preview/editors/vscode/package.json:257]: we didn't bundle them at all, which won't work if preview extension shares `node_modules` with the workspace.

image

File: contrib/typst-preview/editors/vscode/package.json [hunk 4]
```diff
@@ -16,8 +16,10 @@
   "categories": [
     "Other"
   ],
-  "activationEvents": [],
-  "main": "./out/src/extension.js",
+  "activationEvents": [
+    "onLanguage:typst"
```
[the remark this answers is not in the document]
Comment (Myriad-Dreamin) [reply contrib/typst-preview/editors/vscode/package.json:20]: We can pick the patch just before we cleaning up all PRs and removing the code in preview repo.

File: contrib/typst-preview/editors/vscode/package.json (hunk 3 above)
[the remark this answers is not in the document]
Comment (Enter-tainer) [reply contrib/typst-preview/editors/vscode/package.json:257]: okay i think there is some magical things. I just notice things like

```
  "files": ["../../../../editors/vscode/src/extension.preview.ts"],
  "include": ["../../../../editors/vscode/src/**/*.ts"]
```

**Pull request merged**